Staging Sync: Remove effect that was causing check and uncheck without user intervention

epeicher opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Resolves DOTCOM-14119

## Proposed Changes

* Remove `useEffect` that was causing the check and uncheck of the checkboxes without user intervention
* Enable the button so the user can do a full synchronization as long as the domain confirmation passes
* Fix an existing issue with the domain confirmation not correctly populated in some scenarios
* If there is no backup, the checkboxes will be selected.
* While reading the endpoint that identifies if there is a backup or not. A loading state will be displayed with a skeleton placeholder, instead of the checkboxes.
* When closing the modal, all the checkboxes will be unchecked 
* `Database tables` will be renamed to `Database` because is shorter and clear.

|Backup information loading | Backup don't exist| Backup exists|
|-------|------|------|
|  <img width="1350" height="1402" alt="CleanShot 2025-08-08 at 17 03 39@2x" src="https://github.com/user-attachments/assets/436060ce-f976-4824-8b90-6c925fbd7ab8" />|  <img width="1350" height="1402" alt="CleanShot 2025-08-08 at 17 03 52@2x" src="https://github.com/user-attachments/assets/331994f4-9ac5-4528-8d1f-3799c1f4837a" />|<img width="1350" height="1402" alt="CleanShot 2025-08-08 at 17 04 03@2x" src="https://github.com/user-attachments/assets/a5f2e3d8-1db4-432e-bee3-35d969696f53" />|


## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* To fix the bug linked in DOTCOM-14119

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Apply these changes or use Calypso live link below
* Navigate to a site that does not have a staging site and add a staging site to it 
* Wait for the staging site to be created, and just when it is created, select `Pull from Staging` in the `Sync` dropdown on the top right
* Check that temporarily, while the backup endpoint is loading, there is a skeleton displayed.
* After the backup endpoint completes, check the checkboxes are selected and disabled, and that the `All files and folders` dropdown on the right-hand side of the modal is disabled.
* Check that you cannot see any unwanted effect of checkboxes being selected and unselected when opening the `Sync` modal as reported in the video in the linked task DOTCOM-14119
* Check that a full `Pull` can be done
* Check that when selecting/unselecting the checkboxes and closing the modal, you open it again and they are unchecked always.

### Testing intermediate states
Additionally to the user testing above, you can test the different states by uncommenting the following flags:
```diff
diff --git i/client/sites/staging-site/components/staging-site-sync-modal/index.tsx w/client/sites/staging-site/components/staging-site-sync-modal/index.tsx
index c0273ea82bc..c6ad1e533c5 100644
--- i/client/sites/staging-site/components/staging-site-sync-modal/index.tsx
+++ w/client/sites/staging-site/components/staging-site-sync-modal/index.tsx
@@ -285,11 +285,16 @@ export default function SyncModal( {
 		},
 	} );
 
-	const { backupAttempt: lastKnownBackupAttempt, isLoading: isLoadingBackupAttempt } =
+	let { backupAttempt: lastKnownBackupAttempt, isLoading: isLoadingBackupAttempt } =
 		useFirstMatchingBackupAttempt( querySiteId, {
 			sortOrder: 'desc',
 			successOnly: true,
 		} );
+	// simulate a non-existing backup
+	lastKnownBackupAttempt = null;
+
+	// simulate a loading state
+	// isLoadingBackupAttempt = true;
 	const rewindId = lastKnownBackupAttempt?.rewindId;
 
 	const shouldDisableGranularSync = ! lastKnownBackupAttempt && ! isLoadingBackupAttempt;

```

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

epeicher committed:
Staging Sync: Unselect nodes by default in all cases

gcsecsey reviewed (approved):
Thanks @epeicher for making this more robust, great work! The changes make sense to me and I think this is now working correctly:
- I'm unable to select anything when the site doesn't yet have a backup
- I can select nodes when there's a backup, but by default everything's unchecked

https://github.com/user-attachments/assets/a125c998-eba8-4f39-b8a8-e81a25dd6803

Non-blocking: Now that the dropdown for specific files is disabled, there's no way for the user to see the message that there are no backups of the site yet. I think we should move that message out of the specific files and folders section to make it work with this change.

katinthehatsite commented:
> Now that the dropdown for specific files is disabled, there's no way for the user to see the message that there are no backups of the site yet. I think we should move that message out of the specific files and folders section to make it work with this change.

I was going to mention the same thing. It seems that now the user can not select anything but they are also missing information on why they can not select anything.

epeicher committed:
If there is no backup, check nodes and disable them as the nodes do not exist

epeicher commented on client/sites/staging-site/components/staging-site-sync-modal/index.tsx:
```diff
@@ -527,6 +512,7 @@ export default function SyncModal( {
 								</HStack>
 							}
 							onChange={ handleDomainConfirmation }
+							value={ domainConfirmation }
```
This is an existing issue where the input was not always correctly populated, not directly related to these changes

epeicher commented:
@katinthehatsite, @gcsecsey, @wojtekn, I have added latest changes here [7ec915c](https://github.com/Automattic/wp-calypso/pull/105082/commits/7ec915c10a12bd44b54c6da82b9a721006f5a0a4) and updated the title and description of the PR. Please let me know your view on the changes. Thanks!

katinthehatsite reviewed (approved):
The changes work as expected for me 👍 

Non-blocking: One thing that I noticed  is that when you select check the checkboxes and then close the modal and re-open it again, the boxes that you checked remain checked. Not sure if we should reset the state of the modal when it closes in that casse.

gcsecsey commented:
> > Non-blocking: One thing that I noticed is that when you select check the checkboxes and then close the modal and re-open it again, the boxes that you checked remain checked. Not sure if we should reset the state of the modal when it closes in that casse.
> 
> That is a very good point that I have considered a lot, but I don't have a clear option. That's why I have left the default behaviour. I would be very interested in others' opinion @wojtekn, @gcsecsey: On one hand, it is nice that the user selects something, then closes the modal, and then the same selection persists when opening the modal again. On the other hand, the system can always reset all selections when closing the modal and all checkboxes will be displayed as unchecked, I think we can add some code to do that on modal closing. What do you think?

epeicher commented:
Ok, after a conversation p1754651006313789/1754650370.570019-slack-C04GESRBWKW we have decided the following:

- If there is no backup, the checkboxes will be selected and clearly disabled so the user identifies that they cannot be checked. The tooltip will inform the user about the reason.
- The tooltip will be displayed in all the disabled area, which is the `Files and folders` and `Database tables` rows.
- While reading the endpoint that identifies if there is a backup or not. A loading state will be displayed with a skeleton placeholder or similar (instead of the checkboxes), so the user doesn't find states changing from check to uncheck temporarily.
- When having a backup: When closing the modal, all the checkboxes will be reset to uncheck so the user will always see unchecked checkboxes when opening the modal. This applies to the main use case of having a backup.
- `Database tables` will be renamed to `Database` to better resemble Studio

epeicher committed:
Unselect everything when closing the modal

epeicher committed:
Add a placeholder when loading the backup information

epeicher committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into dotcom-14119-sync-modal-checkboxes-are-checked-when-pull-from-staging

epeicher committed:
Do not show warnings when loading backup information

katinthehatsite reviewed (approved):
I think this looks good 👍

* the database is correctly renamed
* the state gets reset when you are closing the modal and the boxes get unchecked
* the loading state is present

Once we complete the follow up task, I think this will be 👍

Commit:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into dotcom-14119-sync-modal-checkboxes-are-checked-when-pull-from-staging

wojtekn commented:
I see one more thing - shouldn't checkboxes have a styling showing their disabled state? They are disabled, but still look like enabled elements:

<img width="635" height="173" alt="Screenshot 2025-08-11 at 12 08 59" src="https://github.com/user-attachments/assets/36e1b60f-9b50-4355-b0ef-09e1d7c24628" />

It seems style for disabled element is there, but's it's overwritten:

<img width="574" height="270" alt="Screenshot 2025-08-11 at 12 10 40" src="https://github.com/user-attachments/assets/d374bc61-a8df-4f21-a922-06df5cf32aac" />

katinthehatsite commented:
@wojtekn I will handle it in https://linear.app/a8c/issue/DOTCOM-14165/staging-sites-redesign-update-tooltip-and-checkbox-disable-style along with tasks outlined there
